src/visual.ts

```diff
@@ -525,114 +563,128 @@ export class DualKpi implements IVisual {
+                    valueAsPercent: data.bottomValueAsPercent,
+                    width: chartWidth,
+                    position: DualKpiChartPositionType["bottom"],
+                    showTextOverlay: data.settings.dualKpiValues.show.value,
+                    showDefaultTextOverlay: !data.settings.dualKpiValues.showKpiValuesBottom.value,
+                    defaultTextOverlay: data.settings.dualKpiValues.bottomChartDefaultKpiValue.value,
+                    fontSizeAutoFormattingTitle: data.settings.dualKpiTitleFormatting.fontSizeAutoFormatting.value,
+                    fontSizeAutoFormattingValue: data.settings.dualKpiValueFormatting.fontSizeAutoFormatting.value,
+                    titleFontSize: data.settings.dualKpiTitleFormatting.font.fontSize.value,
+                    valueFontSize: data.settings.dualKpiValueFormatting.font.fontSize.value,
+                    isBoldTitle: data.settings.dualKpiTitleFormatting.font.bold.value,
+                    isBoldValue: data.settings.dualKpiValueFormatting.font.bold.value,
+                    isItalicTitle: data.settings.dualKpiTitleFormatting.font.italic.value,
+                    isItalicValue: data.settings.dualKpiValueFormatting.font.italic.value,
+                    isUnderlineTitle: data.settings.dualKpiTitleFormatting.font.underline.value,
+                    isUnderlineValue: data.settings.dualKpiValueFormatting.font.underline.value,
+                    isUpperCasedTitle: data.settings.dualKpiTitleFormatting.upperCase.value,
+                    fontFamilyTitle: data.settings.dualKpiTitleFormatting.font.fontFamily.value,
+                    fontFamilyValue: data.settings.dualKpiValueFormatting.font.fontFamily.value,
+                });
+            }
+        } else {
+            this.chartGroupBottom.group.classed(DualKpi.INVISIBLE, true);
+            this.chartGroupBottom.hoverDataContainer.container.classed(DualKpi.INVISIBLE, true);
+        }
+    }
 
-        this.displayRootElement(wasDataSetRendered);
+    private drawTopChart({
+        data,
+        chartHeight,
+        chartWidth,
+        chartSpaceBetween,
+    }: {
+        data: IDualKpiData;
+        chartHeight: number;
+        chartWidth: number;
+        chartSpaceBetween: number;
+    }) {
+        if (data.settings.dualKpiProperties.topChartShow.value) {
+            const topChartAxisConfig = { min: data.settings.dualKpiAxis.topChartAxisMin.value, max: data.settings.dualKpiAxis.topChartAxisMax.value };
+            const topChartPercentChangeStartPoint = DualKpi.getPercentChangeStartPoint(data.topValues, data.topPercentCalcDate);
+ 
 
-        if (data.settings.dualKpiProperties.topChartShow) {
             this.chartGroupTop.group.classed(DualKpi.INVISIBLE, false);
             this.chartGroupTop.hoverDataContainer.container.classed(DualKpi.INVISIBLE, true);
 
             if (data.topValues.length > 0) {
                 this.drawChart({
```

Comment:
move such a big argument object to a variable outside, to make the function itself clearer and easier to read.